### PR TITLE
gpu: Use SetDeviceLoaderData on command buffer

### PR DESCRIPTION
**Pull request by Tony-LunarG**
Use the loader supplied method of initializing loader data

### CODE REVIEW

File: layers/gpu_validation.cpp [hunk 1]
```diff
@@ -254,6 +254,10 @@ void CoreChecks::GpuPostCallRecordCreateDevice(const CHECK_ENABLED *enables) {
         gpu_validation_state->aborted = true;
         return;
     }
+
+    auto chain_info = get_chain_info(pCreateInfo, VK_LOADER_DATA_CALLBACK);
```
Comment (jzulauf-lunarg) [inline layers/gpu_validation.cpp]: Add comment that this is guaranteed to work in 1.1. @cnorthrop also true on Android loader?

Comment (jzulauf-lunarg) [reply layers/gpu_validation.cpp]: and assert

Comment (Tony-LunarG) [reply layers/gpu_validation.cpp]: SetDeviceLoaderData was added to the Android loader in 2016.  Vulkan 1.1 launched in 2018, so it's safe to assume that SetDeviceLoaderData will be there for an Android 1.1 loader

Comment (jzulauf-lunarg) [reply layers/gpu_validation.cpp]: Add a comment then....

**Pull request merged**